src/qt3utils/applications/controllers/utils.py

```diff
@@ -0,0 +1,380 @@
+) -> Tuple[ttk.Label, ttk.Checkbutton, tk.BooleanVar]:
+    """
+    This helper method creates a row of the label-check button tk widgets.
+
+    Parameters
+    ----------
+    parent: Union[tk.Toplevel, ttk.Widget]
+        The parent window the label-check button will be inserted in.
+    label_text: str
+        The text of the label.
+    row: int
+        The row of label-check button in the window.
+    value: bool
+        The initial value of the check button.
+    label_padx: int
+        The padding of the label. Default is 10.
+
+    Returns
+    -------
+    Tuple[ttk.Label, ttk.Checkbutton, tk.BoolVar]
+        The generated label, checkbutton and variable.
+    """
+    label = ttk.Label(parent, text=label_text)
+    label.grid(row=row, column=0, padx=label_padx)
+
+    variable = tk.BooleanVar(value=value)
+
+    tick_button = ttk.Checkbutton(parent, variable=variable)
+    tick_button.grid(row=row, column=1)
+
+    return label, tick_button, variable
+
+
+def make_separator(
+        parent: Union[tk.Toplevel, ttk.Widget],
+        row: int,
+        orient: Literal['horizontal', 'vertical'] = tk.HORIZONTAL,
+        column: int = 0,
+        column_span: int = 2,
+        padx: int = 10,
+        pady: int = 5,
+        sticky: str = tk.EW,
+) -> ttk.Separator:
+    """
+    This helper method creates a separator.
+
+    Parameters
+    ----------
+    parent: Union[tk.Toplevel, ttk.Widget]
+        The parent window the separator will be inserted in.
+    row: int
+        The row of separator in the window.
+    orient: Literal['horizontal', 'vertical']
+        The orientation of the separator. Default is 'horizontal'.
+    column: int
+        The column of separator in the window. Default is 0.
+    column_span: int
+        The column span of separator in the window. Default is 2.
+    padx: int
+        The padding of the separator in the x direction. Default is 10.
+    pady: int
+        The padding of the separator in the y direction. Default is 5.
+    sticky: str
+        The sticky direction of the separator. Default is 'EW'.
+
+    Returns
+    -------
+    ttk.Separator
+        The generated separator.
+    """
+    separator = ttk.Separator(parent, orient=orient)
+    separator.grid(row=row, column=column, columnspan=column_span, sticky=sticky, padx=padx, pady=pady)
+
+    return separator
+
+
+def prepare_list_for_option_menu(list_to_prepare: Sequence[Any], filler_value: str = 'None') -> List[str]:
+    """
+    This helper method prepares a list for the option menu.
+
+    Parameters
+    ----------
+    list_to_prepare: Sequence[Any]
+        The list to prepare.
+    filler_value: str
+        This value will be added to the returned list if the list is empty.
+
+    Returns
+    -------
+    List[str]
+        The prepared list.
+    """
+    return [str(element) for element in list_to_prepare] if len(list_to_prepare) > 0 else [filler_value]
+
+
+def make_popup_window_and_take_threaded_action(
```

Comment:
I like the current implementation and this might be overkill but maybe we can implement further threading safeguards in order to make sure that there is thread safety. I say this because I could see us potentially running into issues if the system doesn't like what the user selects. Maybe something like a queue, will need to think about it more.

Reply:
Sounds good. We can look into it in a future PR